### PR TITLE
Remove the need for `getConfig` by centralising logic surrounding finding the config file for given args

### PR DESCRIPTION
**Pull request by andyjessop**
Fixes #000.

There was a lot of logic spread around relating to how to locate the config path. Whether to use `args.config`, `args.script`, some specified `configPath`, etc. If we centralise this logic inside `readConfig`, we can remove a lot of the complexity.

In fact, it removes entirely the need for the first argument to `readConfig`, so in most cases, the call now just reads `readConfig(args)`.

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: already covered
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: internal only

### CODE REVIEW

File: packages/wrangler/src/cloudchamber/common.ts [hunk 1]
```diff
@@ -97,19 +97,19 @@ export function handleFailure<
 		? K
 		: never,
 >(
-	cb: (t: CommandArgumentsObject, config: Config) => Promise<void>
+	cb: (args: CommandArgumentsObject, config: Config) => Promise<void>
```
Comment (andyjessop) [inline packages/wrangler/src/cloudchamber/common.ts:100]: Some housekeeping 😁

File: packages/wrangler/src/config/config-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import path from "path";
+import { findUpSync } from "find-up";
+
+/**
+ * Resolve the path to the configuration file, given the `config` and `script` optional command line arguments.
+ * `config` takes precedence, then `script`, then we just use the cwd.
+ */
+export function resolveWranglerConfigPath({
```
Comment (andyjessop) [inline packages/wrangler/src/config/config-helpers.ts:8]: This function is the meat of the PR, centralising the logic surrounding `config`, `script`, and the default option.

File: packages/wrangler/src/index.ts [hunk 3]
```diff
@@ -1109,7 +1109,7 @@ export async function main(argv: string[]): Promise<void> {
 		// key to fetch) or flags
 
 		try {
-			const configPath = args.config ?? findWranglerConfig(process.cwd());
+			const configPath = resolveWranglerConfigPath(args);
 			const rawConfig = readRawConfig(args.config);
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:1113]: This logic doesn't look right (not a change in your PR). I _think_ we should be passing `configPath` to `readRawConfig`? `args.config` won't always be defined

File: packages/wrangler/src/triggers/index.ts [hunk 4]
```diff
@@ -57,7 +57,7 @@ async function triggersDeployHandler(
 ) {
 	await printWranglerBanner();
 
-	const config = readConfig(undefined, args);
+	const config = readConfig(args);
```
Comment (penalosa) [inline packages/wrangler/src/triggers/index.ts:60]: This is a change in behaviour, but it looks like the previous behaviour must've been a bug

Comment (andyjessop) [reply packages/wrangler/src/triggers/index.ts:60]: Oh yeah, you're right. Didn't spot that.

**Pull request merged**